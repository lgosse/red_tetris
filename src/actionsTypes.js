// New Game actions
export const NEW_GAME_CREATE = 'NEW_GAME_CREATE';
export const NEW_GAME_JOIN = 'JOIN_GAME';

// PartyList actions
export const PARTY_LIST = 'server/party-list';
export const RESPONSE_PARTY_LIST = 'RESPONSE_PARTY_LIST';

// Party actions
export const PARTY_ADD = 'server/party-add';
export const PARTY_UPDATE = 'PARTY_UPDATE';
export const PARTY_OPEN = 'server/toggle-open-party';
export const PARTY_JOIN = 'server/party-join';
export const PARTY_LEAVE = 'server/party-leave';
export const PARTY_LEFT = 'PARTY_LEFT';
export const PARTY_KICK_PLAYER = 'server/party-kick-player';
export const PARTY_START = 'server/party-toggle-playing';
export const PARTY_SEND_MESSAGE = 'server/send-message';
export const PARTY_RECEIVE_MESSAGE = 'PARTY_RECEIVE_MESSAGE';

//Music action
export const TOGGLE_MUSIC = 'TOGGLE_MUSIC';

// Player actions
export const PLAYER_UPDATE = 'PLAYER_UPDATE';
export const PLAYER_GET = 'PLAYER_GET';
export const PLAYER_TOGGLE_READY = 'server/player-toggle-ready';

export const RANKINGS_LIST = 'server/ranking-list';
export const RESPONSE_RANKINGS_LIST = 'RESPONSE_RANKINGS_LIST';

// Game actions
export const GAME_PIECES_PIECE_ROTATE_SERVER =
  'GAME_PIECES_PIECE_ROTATE_SERVER';
export const GAME_PIECES_PIECE_ROTATE_SUCCESS =
  'GAME_PIECES_PIECE_ROTATE_SUCCESS';
export const GAME_PIECES_PIECE_MOVE_SERVER = 'GAME_PIECES_PIECE_MOVE_SERVER';
export const GAME_PIECES_PIECE_MOVE_SUCCESS = 'GAME_PIECES_PIECE_MOVE_SUCCESS';
export const GAME_PIECES_CLAIM_PIECE = 'server/claim-piece';
export const GAME_PIECES_CLAIM_PIECE_SUCCESS = 'PLAYER_CLAIM_PIECE_SUCCESS';
export const GAME_BONUS_ADD = 'GAME_BONUS_ADD';
export const GAME_MALUS_ADD = 'GAME_MALUS_ADD';
export const GAME_MALUS_ADD_SUCCESS = 'GAME_MALUS_ADD_SUCCESS';
export const GAME_PIECES_UPDATE = 'GAME_PIECES_UPDATE';
export const GAME_PIECE_UPDATE = 'GAME_PIECE_UPDATE';

export const GAME_HAS_FOCUS = 'GAME_HAS_FOCUS';
export const GAME_LOSE_FOCUS = 'GAME_LOSE_FOCUS';
export const GAME_BOARD_DELETE_LINES = 'GAME_PIECES_DELETE_LINES';
export const GAME_BOARD_BLOCK_LINES_SERVER = 'GAME_BOARD_BLOCK_LINES_SERVER';
export const GAME_BOARD_BLOCK_LINES_SUCCESS = 'GAME_BOARD_BLOCK_LINES_SUCCESS';
export const GAME_BOARD_NOTIFY_GRID_UPDATE = 'server/notify-grid-update';
export const GAME_BOARD_UPDATE = 'GAME_BOARD_UPDATE';

export const GAME_LOSE = 'server/game-lose';
export const GAME_END = 'GAME_END';
export const GAME_DISPLAY_END = 'GAME_DISPLAY_END';
export const GAME_HIDE_END = 'GAME_HIDE_END';
export const GAME_MODS_SET = 'GAME_MODS_SET';

export const GAME_SCORE_UPDATE = 'GAME_SCORE_UPDATE';
export const GAME_SCORE_RESET = 'GAME_SCORE_RESET';

export const INPUT_KEYBOARD = 'INPUT_KEYBOARD';
export const REMOVE_INPUT_KEYBOARD = 'REMOVE_INPUT_KEYBOARD';

// Server actions
export const SERVER_PING = 'server/ping';
export const SERVER_PING_USER = 'server/pingUser';
export const SERVER_PONG_USER = 'server/pongUser';

// Router actions
export const LOCATION_CHANGE = '@@router/LOCATION_CHANGE';

// Alert actions
export const ALERT_POP = 'ALERT_POP';
export const ALERT_RESET = 'ALERT_RESET';
