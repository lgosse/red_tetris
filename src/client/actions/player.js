import {
  PLAYER_GET,
  PLAYER_SAVE,
  PLAYER_UPDATE,
  PLAYER_PIECE_ROTATE,
  PLAYER_PIECE_MOVE,
  PLAYER_DELETE_LINES,
  PLAYER_CLAIM_PIECE,
  PLAYER_CLAIM_PIECE_SUCCESS
} from '../../actionsTypes';

export const getPlayer = () => ({
  type: PLAYER_GET
});

export const updatePlayer = player => ({
  type: PLAYER_UPDATE,
  player
});

export const savePlayer = player => ({
  type: PLAYER_SAVE,
  player
});

export const rotatePiece = (player, direction) => ({
  type: PLAYER_PIECE_ROTATE,
  player,
  direction
});

export const claimPiece = partyId => ({
  type: PLAYER_CLAIM_PIECE,
  partyId
});

export const claimPieceSuccess = pieces => ({
  type: PLAYER_CLAIM_PIECE_SUCCESS,
  pieces
});

export const movePiece = direction => ({
  type: PLAYER_PIECE_MOVE,
  direction
});

export const deleteLines = () => ({
  type: PLAYER_DELETE_LINES
});
