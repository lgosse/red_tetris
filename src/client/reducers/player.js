import {
  PLAYER_UPDATE,
  PLAYER_SAVE,
  PLAYER_GET,
  PLAYER_PIECE_ROTATE,
  PLAYER_PIECE_MOVE,
  PLAYER_DELETE_LINES,
} from '../../actionsTypes';

const testCollision = (piece, grid) => {
  let collisionLocation = 0;
  let collision = false;
  piece.grid.forEach((line, y) => {
    line.forEach((col, x) => {
      if (col !== 0) {
        let pos = {
          x: x + piece.x,
          y: y + piece.y,
        };
        if (pos.x < 0 || pos.x >= grid[0].length || pos.y < 0 || pos.y >= grid.length || grid[pos.y][pos.x] !== 0) {
          if (x <= parseInt((piece.grid.length - 1) / 2)) collisionLocation--;
          if (x >= parseInt(piece.grid.length / 2)) collisionLocation++;
          collision = true;
        }
      }
    });
  });
  return { collide: collision, location: collisionLocation };
};

const checkLines = grid => {
  let lines = [];
  grid.forEach((line, y) => {
    let i = 0;
    line.forEach((col, x) => {
      if (col !== 0) i++;
    });
    if (i === grid[0].length) {
      lines.push(y);
    }
  });
  if (lines.length === 0) return null;
  return lines;
};

const deleteLines = (grid, lines) => {
  let newGrid = [];
  let y;
  let cur = grid.length - 1;
  for (y = grid.length - 1; y >= 0; y--) {
    let i;
    if ((i = lines.indexOf(y)) === -1) {
      newGrid[cur] = [...grid[y]];
      lines[i] = -1;
      cur--;
    }
  }
  while (cur >= 0) {
    newGrid[cur] = [];
    for (y = 0; y < grid[0].length; y++) {
      newGrid[cur].push(0);
    }
    cur--;
  }
  console.log(newGrid);
  return newGrid;
};

const calcWeight = grid => {
  let weight = 0;
  grid.forEach((line, y) => {
    line.forEach((col, x) => {
      if (col !== 0) weight += x - parseInt(grid.length / 2);
    });
  });
  if (weight > 0) weight = -1;
  else if (weight < 0) weight = 1;
  return weight;
};

const gridFusion = (piece, grid) => {
  let newGrid = [...grid];
  piece.grid.forEach((line, y) => {
    line.forEach((col, x) => {
      if (col != 0) {
        if (y + piece.y < 0) return null;
        newGrid[y + piece.y][x + piece.x] = col;
      }
    });
  });
  return newGrid;
};

const findPlace = (piece, grid, dir) => {
  if (dir > piece.grid.length / 2 || dir < -piece.grid.length / 2) return null;
  let test;
  if ((test = testCollision(piece, grid)).collide) {
    let pos;
    if (dir < 0 || (dir === 0 && test.location > 0)) {
      pos = findPlace({ ...piece, x: piece.x - 1 }, grid, dir - 1);
    } else if (dir > 0 || (dir === 0 && test.location < 0)) pos = findPlace({ ...piece, x: piece.x + 1 }, grid, dir + 1);
    else {
      let weight;
      if (dir == 0 && (weight = calcWeight(piece.grid)) != 0) {
        test = testCollision({ ...piece, x: piece.x + weight }, grid);
        if (test.collide) pos = null;
        else pos = { x: piece.x + weight };
      } else pos = null;
    }
    return pos;
  } else {
    return {
      x: piece.x,
    };
  }
};

const getPlayer = () => {
  const playerItem = localStorage.getItem('player');
  if (playerItem) return playerItem;

  return '';
};

const savePlayer = action => {
  localStorage.setItem('player', action.player.nickname);
};

const initPlayer_test = () => {
  let playert = { nickname: 'thomas' };
  playert.piece = {
    grid: [[5, 5, 0], [0, 5, 5], [0, 0, 0]],
    grid2: [[0, 1, 0, 0], [0, 1, 0, 0], [0, 1, 0, 0], [0, 1, 0, 0]],
    x: 5,
    y: 5,
    end: false,
  };

  // player.piece.grid = Pieces.rotate(player.piece.grid, -1);
  playert.grid = [
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 4],
    [0, 0, 0, 0, 0, 0, 1, 0, 0, 4],
    [0, 0, 0, 0, 0, 0, 1, 0, 0, 4],
    [0, 1, 0, 0, 1, 1, 1, 2, 2, 4],
    [1, 1, 1, 0, 1, 1, 1, 0, 2, 2],
  ];
  return playert;
};

const gridZero = size => {
  let grid = [];
  let i, j;
  for (i = 0; i < size; i++) {
    grid[i] = [];
    for (j = 0; j < size; j++) {
      grid[i][j] = 0;
    }
  }
  return grid;
};

const player = (state = initPlayer_test(), action) => {
  switch (action.type) {
  case PLAYER_UPDATE:
    return {
      ...state,
      ...action.player,
    };
  case PLAYER_SAVE: {
    savePlayer(action);
    return state;
  }
  case PLAYER_GET:
    return {
      ...state,
      nickname: getPlayer(),
    };

  case PLAYER_PIECE_ROTATE: {
    let newGrid = gridZero(action.player.piece.grid.length);
    action.player.piece.grid.forEach((line, y) => {
      line.forEach((col, x) => {
        if (col != 0) {
          newGrid[y + (action.player.piece.grid.length - 1) * ((1 - action.direction) / 2) + action.direction * x - y][
              x + (action.player.piece.grid.length - 1) * ((action.direction + 1) / 2) - x - y * action.direction
            ] = col;
        }
      });
    });
    const pos = findPlace({ ...state.piece, grid: newGrid }, state.grid, 0);
    if (pos === null) {
      return state;
    } else {
      return {
        ...state,
        piece: {
          ...state.piece,
          grid: newGrid,
          ...pos,
        },
      };
    }
  }

  case PLAYER_DELETE_LINES: {
    console.log('OK');
    if (state.lines !== null) {
      const newGrid = deleteLines(state.grid, state.lines);
      return {
        ...state,
        grid: newGrid,
        lines: null,
      };
    }
    return state;
  }

  case PLAYER_PIECE_MOVE: {
    const pos = {
      x: state.piece.x + action.direction,
      y: action.direction === 0 ? state.piece.y + 1 : state.piece.y,
    };
    const res = testCollision({ ...state.piece, ...pos }, state.grid);
    if (res.collide) {
      if (action.direction === 0) {
        let newGrid = gridFusion(state.piece, state.grid);
        let lines = checkLines(newGrid);

          /*
        let newPiece = {
          grid: [[5, 5, 0], [0, 5, 5], [0, 0, 0]],
          x: 5,
          y: 0,
        }; piece: claimPiece();
          */
        if (newGrid === null) return { ...state, piece: null, lines, ending: true };

          // testCollision(newPiece, newGrid).collide
        return {
          ...state,
          grid: newGrid,
          piece: null,
          lines,
        };
      } else return state;
    } else {
      return {
        ...state,
        piece: {
          ...state.piece,
          ...pos,
        },
      };
    }
  }

  default:
    return state;
  }
};

export default player;
