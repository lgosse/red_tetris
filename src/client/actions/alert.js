import { ALERT_POP } from "../../actionsTypes";

export const alert = message => {
  return {
    type: ALERT_POP,
    message
  };
};
