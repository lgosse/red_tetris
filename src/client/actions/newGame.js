import { NEW_GAME_CREATE, NEW_GAME_JOIN } from "../../actionsTypes";

export const newGame = infos => {
  return {
    type: NEW_GAME_CREATE,
    infos
  };
};

export const joinGame = infos => {
  return {
    type: NEW_GAME_JOIN,
    infos
  };
};
