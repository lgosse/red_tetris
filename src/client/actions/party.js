import {PARTY_GET, PARTY_SAVE, PARTY_ADD, PARTY_UPDATE, PARTY_JOIN} from "../../actionsTypes";

export const getParty = () => {
  return {
    type: PARTY_GET
  };
};

export const saveParty = party => {
  return {
    type: PARTY_SAVE,
    party
  };
};

export const addParty = party => {
  return {
    type: PARTY_ADD,
    party
  };
};

export const joinParty = (party, player) => {
  if (party.players.length < party.size) {
    return {
      type: PARTY_JOIN,
      party,
      player
    };
  }
  else {
    return {

    }
  }
};

export const updateParty = party => {
  return {
    type: PARTY_UPDATE,
    party
  };
};
