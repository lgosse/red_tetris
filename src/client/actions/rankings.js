import { RANKINGS_LIST, RESPONSE_RANKINGS_LIST } from '../../actionsTypes';

export const getRankings = () => ({
  type: RANKINGS_LIST
});

export const getRankingsSuccess = rankings => ({
  type: RESPONSE_RANKINGS_LIST,
  rankings
});
