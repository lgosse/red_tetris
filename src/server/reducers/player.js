import GameModel from '../models/Game';
import { PLAYER_TOGGLE_READY } from '../../actionsTypes';
import { updatePlayer } from '../../client/actions/player';
import { updateParty } from '../../client/actions/party';

const player = async (action, io, socket) => {
  switch (action.type) {
    case PLAYER_TOGGLE_READY: {
      let party;
      try {
        party = await GameModel.findById(socket.partyId).exec();
      } catch (e) {
        console.error(e);

        return;
      }
      if (!party) {
        console.error(`No party found for ID: ${socket.partyId}`);

        return;
      }

      const player = party.getPlayerBySocketId(socket.id);
      player.ready = !player.ready;
      party.updatePlayer(player);

      try {
        await party.save();
        io
          .to(socket.partyId)
          .emit('action', updateParty({ players: party.players }));
        socket.emit('action', updatePlayer(player));
      } catch (error) {
        console.error(error);
      }

      break;
    }

    default:
      break;
  }
};

export default player;
