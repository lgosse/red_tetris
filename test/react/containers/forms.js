import chai from 'chai';
import { expect } from 'chai';
import equalJSX from 'chai-equal-jsx';
import React from 'react';
import { shallow } from 'enzyme';
import sinon, { assert } from 'sinon';

// PlayerForm
import {
  PlayerForm,
  mapStateToPlayerFormProps,
  mapDispatchToPlayerFormProps
} from '../../../src/client/containers/forms/PlayerForm';

import { PLAYER_UPDATE, PLAYER_SAVE } from '../../../src/actionsTypes';

import { updatePlayer, savePlayer } from '../../../src/client/actions/player';

describe('PlayerForm', () => {
  const PLAYER = {
    nickname: 'test'
  };
  describe('PlayerForm component', () => {
    it('should render as expected', () => {
      const output = shallow(<PlayerForm player={{ nickname: undefined }} />);
      output.should.matchSnapshot();
    });
    it('should trigger changeNickname on typing into the input', done => {
      const event = { target: { value: 'yolo' } };
      const changeNickname = (event, player) => {
        event.should.equal(event);
        done();
      };
      const output = shallow(
        <PlayerForm player={PLAYER} changeNickname={changeNickname} />
      );
      output.find('#nicknameInput').simulate('change', event);
    });
  });
  describe('mapStateToPlayerProps', () => {
    it('should map player to PlayerForm props', () => {
      const state = { player: PLAYER };
      const playerFormProps = mapStateToPlayerFormProps(state);

      playerFormProps.should.be.deep.equal({
        player: PLAYER
      });
    });
  });
  describe('mapDispatchToPlayerFormProps', () => {
    it('should map changeNickname function to PlayerForm props', done => {
      global.document = {
        getElementById() {
          return {
            value: {
              trim() {
                return 'toto';
              }
            }
          };
        }
      };
      const dispatch = action => {
        action.player.should.deep.equal({ nickname: 'toto' });
        done();
      };
      const { changeNickname } = mapDispatchToPlayerFormProps(dispatch);
      changeNickname(dispatch);
    });
  });
});

describe('PartyForm', () => {
  describe('PartyForm', () => {
    it('should render as expected', () => {});
  });
  describe('mapState', () => {});
});
