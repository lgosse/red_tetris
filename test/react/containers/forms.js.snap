// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerForm PlayerForm component should render as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayerForm
    player={
      Object {
        "nickname": "test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div
          center={true}
          color="accent"
          gameFont={true}
          padding="20px"
          size="20px"
        >
          Enter your player nickname
        </styled.div>,
        <styled.div
          flex={true}
        >
          <styled.div />
          <styled.div>
            <styled.input
              id="nicknameInput"
              maxLength="10"
              name="nickname"
              onChange={[Function]}
              placeholder="Nickname..."
              value="test"
            />
          </styled.div>
          <styled.div />
        </styled.div>,
      ],
      "padding": "20px",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "center": true,
          "children": "Enter your player nickname",
          "color": "accent",
          "gameFont": true,
          "padding": "20px",
          "size": "20px",
        },
        "ref": null,
        "rendered": "Enter your player nickname",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div />,
            <styled.div>
              <styled.input
                id="nicknameInput"
                maxLength="10"
                name="nickname"
                onChange={[Function]}
                placeholder="Nickname..."
                value="test"
              />
            </styled.div>,
            <styled.div />,
          ],
          "flex": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <styled.input
                id="nicknameInput"
                maxLength="10"
                name="nickname"
                onChange={[Function]}
                placeholder="Nickname..."
                value="test"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "nicknameInput",
                "maxLength": "10",
                "name": "nickname",
                "onChange": [Function],
                "placeholder": "Nickname...",
                "value": "test",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div
            center={true}
            color="accent"
            gameFont={true}
            padding="20px"
            size="20px"
          >
            Enter your player nickname
          </styled.div>,
          <styled.div
            flex={true}
          >
            <styled.div />
            <styled.div>
              <styled.input
                id="nicknameInput"
                maxLength="10"
                name="nickname"
                onChange={[Function]}
                placeholder="Nickname..."
                value="test"
              />
            </styled.div>
            <styled.div />
          </styled.div>,
        ],
        "padding": "20px",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "center": true,
            "children": "Enter your player nickname",
            "color": "accent",
            "gameFont": true,
            "padding": "20px",
            "size": "20px",
          },
          "ref": null,
          "rendered": "Enter your player nickname",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div />,
              <styled.div>
                <styled.input
                  id="nicknameInput"
                  maxLength="10"
                  name="nickname"
                  onChange={[Function]}
                  placeholder="Nickname..."
                  value="test"
                />
              </styled.div>,
              <styled.div />,
            ],
            "flex": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <styled.input
                  id="nicknameInput"
                  maxLength="10"
                  name="nickname"
                  onChange={[Function]}
                  placeholder="Nickname..."
                  value="test"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "id": "nicknameInput",
                  "maxLength": "10",
                  "name": "nickname",
                  "onChange": [Function],
                  "placeholder": "Nickname...",
                  "value": "test",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
